src/ssh_mux/config.rs

```diff
@@ -15,13 +15,18 @@
 
 use std::process::Command;
 
+/// Returns true if it looks like we should create our own socket to manage SSH connection
+/// multiplexing, false otherwise. Specifically this returns true if the host configuration does
+/// not include `ControlMaster auto`; if it does, then we assume we can just use (or create) the
+/// pre-existing socket.
+///
+/// This is not a perfect heuristic; e.g. if the host has too short a `ControlPersist` value, then
+/// we might wind up starting up multiple connections regardless.
+///
+/// If this function encounters an error, it just returns false. The assumption is that whatever
+/// went wrong reading the config will also go wrong connecting to the host, and that there's no
+/// reason to stand up our own managed socket for a connection we expect to fail anyway.
```

Comment:
I think this could be made a bit clearer
```suggestion
/// Guess if we should create create our own socket or attempt to reuse an existing one.
///
/// This function checks the output of `ssh -G` for the given host and returns false if the user has
/// set a value for the `ControlPersist` directive, which we assume means there's an existing socket
/// we can reuse.
///
/// We don't bother checking the timeout value or errors here, since we will fall back to creating
/// a new socket if the control socket has gone away, and any any relevant errors will be
/// reported later when we attempt to connect.
```